jms/src/main/scala/akka/stream/alpakka/jms/scaladsl/JmsConsumer.scala

```diff
@@ -65,8 +66,10 @@ object JmsConsumer {
    * @param settings The settings for the ack source.
    * @return Source for JMS messages in an AckEnvelope.
    */
-  def ackSource(settings: JmsConsumerSettings): Source[AckEnvelope, KillSwitch] =
-    Source.fromGraph(new JmsAckSourceStage(settings))
+  def ackSource(settings: JmsConsumerSettings): Source[AckEnvelope, KillSwitch] = {
+    require(settings.destination.isDefined, noConsumerDestination(settings))
+    Source.fromGraph(new JmsAckSourceStage(settings, settings.destination.get))
```

Comment:
Ditto here. Using match and extraction is a little safer. If you don't cover all cases, the compiler will complain.